assets/js/modules/reader-revenue-manager/datastore/publications.test.js

```diff
@@ -16,115 +17,202 @@
 import * as fixtures from './__fixtures__';
+import { enabledFeatures } from '../../../features';
+import { CORE_UI } from '../../../googlesitekit/datastore/ui/constants';
 import {
 	MODULES_READER_REVENUE_MANAGER,
+	MODULE_SLUG,
 	PUBLICATION_ONBOARDING_STATES,
+	UI_KEY_SHOW_RRM_PUBLICATION_APPROVED_NOTIFICATION,
 } from './constants';
 
 describe( 'modules/reader-revenue-manager publications', () => {
 	let registry;
 
+	const getModulesEndpoint = new RegExp(
+		'^/google-site-kit/v1/core/modules/data/list'
+	);
+
 	const publicationsEndpoint = new RegExp(
 		'^/google-site-kit/v1/modules/reader-revenue-manager/data/publications'
 	);
 
+	const settingsEndpoint = new RegExp(
+		'^/google-site-kit/v1/modules/reader-revenue-manager/data/settings'
+	);
+
 	beforeAll( () => {
 		API.setUsingCache( false );
 	} );
 
 	beforeEach( () => {
+		enabledFeatures.add( 'rrmModule' ); // Enable RRM module to get its features.
 		registry = createTestRegistry();
+		provideUserInfo( registry );
 	} );
 
-	describe( 'selectors', () => {
-		describe( 'getPublications', () => {
-			it( 'should use a resolver to make a network request', async () => {
-				fetchMock.get( publicationsEndpoint, {
-					body: fixtures.publications,
-					status: 200,
-				} );
+	describe( 'actions', () => {
+		beforeEach( () => {
+			// Make sure the RRM module is active and connected.
+			const extraData = [
+				{
+					slug: MODULE_SLUG,
+					active: true,
+					connected: true,
+				},
+			];
+			provideModules( registry, extraData );
+			provideModuleRegistrations( registry, extraData );
+		} );
 
-				const initialPublications = registry
-					.select( MODULES_READER_REVENUE_MANAGER )
-					.getPublications();
-				expect( initialPublications ).toBeUndefined();
+		describe( 'syncPublicationOnboardingState', () => {
+			it( 'should return undefined when no publication ID is present', async () => {
+				registry
+					.dispatch( MODULES_READER_REVENUE_MANAGER )
+					.receiveGetPublications( fixtures.publications );
 
-				await untilResolved(
-					registry,
-					MODULES_READER_REVENUE_MANAGER
-				).getPublications();
-				expect( fetchMock ).toHaveFetched( publicationsEndpoint );
+				registry
+					.dispatch( MODULES_READER_REVENUE_MANAGER )
+					.receiveGetSettings( {} );
 
-				const publications = registry
-					.select( MODULES_READER_REVENUE_MANAGER )
-					.getPublications();
-				expect( fetchMock ).toHaveFetchedTimes( 1 );
-				expect( publications ).toEqual( fixtures.publications );
-				expect( publications ).toHaveLength(
-					fixtures.publications.length
-				);
+				const syncStatus = await registry
+					.dispatch( MODULES_READER_REVENUE_MANAGER )
+					.syncPublicationOnboardingState();
+
+				expect( syncStatus ).toBeUndefined();
 			} );
 
-			it( 'should not make a network request if publications are already present', async () => {
+			it( 'should update the settings and call the saveSettings endpoint', async () => {
```

Comment:
I think this would be an appropriate test to mock `Date.now()` in, so as to provide a more precise assertions below:

```suggestion
			it( 'should update the settings and call the saveSettings endpoint', async () => {
				const originalDateNow = Date.now;
				// Mock the date to be an arbitrary time.
				const mockNow = new Date( '2020-01-01 12:30:00' ).getTime();
				Date.now = jest.fn( () => mockNow );
```